test/e2e/framework/assertions.go

```diff
@@ -368,6 +368,54 @@ func (f *Framework) AssertOperatorCondition(conditionType configv1.ClusterStatus
 	}
 }
 
+func (f *Framework) AssertOperatorConditionReason(conditionType configv1.ClusterStatusConditionType, conditionReason string) func(t *testing.T) {
+	return func(t *testing.T) {
+		reporter := f.OperatorClient.StatusReporter()
+		err := Poll(time.Second, 5*time.Minute, func() error {
+			co, err := reporter.Get(ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
+			for _, c := range co.Status.Conditions {
+				if c.Type == conditionType {
+					if c.Reason == conditionReason {
+						return nil
+					}
+					return fmt.Errorf("expecting condition %q to have reason %q, got %q", conditionType, conditionReason, c.Reason)
+				}
+			}
+			return fmt.Errorf("failed to find condition %q", conditionType)
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func (f *Framework) AssertOperatorConditionMessage(conditionType configv1.ClusterStatusConditionType, conditionMessage string) func(t *testing.T) {
+	return func(t *testing.T) {
+		reporter := f.OperatorClient.StatusReporter()
+		err := Poll(time.Second, 5*time.Minute, func() error {
+			co, err := reporter.Get(ctx)
+			if err != nil {
+				t.Fatal(err)
```

Comment:
`return err` should suffice, doesn't it?

Reply:
Yes, I had used a similar function, so I've also fixed it